Bug: Segmentation Fault When Creating Advanced Flows

## Bug: Segmentation Fault When Creating Advanced Flows

### Description
`homeyctl flows create --advanced` causes a panic/segmentation fault when attempting to create an advanced flow from a JSON file.

### Steps to Reproduce
1. Create a valid advanced flow JSON file (structure based on existing advanced flows)
2. Run: `homeyctl flows create --advanced flow.json`
3. Observe crash

### Expected Behavior
The advanced flow should be created successfully, similar to how simple flows are created.

### Actual Behavior
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10241cc08]

goroutine 1 [running]:
github.com/langtind/homeyctl/internal/client.(*Client).doRequest(0x0, ...)
	/home/runner/work/homeyctl/homeyctl/internal/client/client.go:41 +0x158
github.com/langtind/homeyctl/internal/client.(*Client).CreateAdvancedFlow(...)
	/home/runner/work/homeyctl/homeyctl/internal/client/client.go:116
```

### Environment
- **homeyctl version**: 0.2.0 (installed via Homebrew)
- **OS**: macOS (ARM64)
- **Homey**: Homey Pro 2023

### Additional Context
The crash occurs in `client.go:41` during `doRequest`, suggesting a nil pointer dereference in the HTTP client initialization or configuration for advanced flows.

### Workaround
Currently using direct API calls to Homey's `/api/manager/flow/flow` endpoint as a workaround.

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -69,7 +69,7 @@ var rootCmd = &cobra.Command{
 		if cmd.Name() == "config" || cmd.Name() == "version" || cmd.Name() == "help" ||
 			cmd.Name() == "set-token" || cmd.Name() == "set-host" || cmd.Name() == "show" ||
 			cmd.Name() == "completion" || cmd.Name() == "ai" || cmd.Name() == "scopes" ||
-			cmd.Name() == "login" || cmd.Name() == "create" || cmdPath == "homeyctl" {
+			cmd.Name() == "login" || cmdPath == "homeyctl token create" || cmdPath == "homeyctl" {
 			return nil
 		}
 

--- a/cmd/root_test.go
+++ b/cmd/root_test.go
@@ -1,0 +1,92 @@
+package cmd
+
+import (
+	"testing"
+)
+
+func TestCommandSkipsConfigLoading(t *testing.T) {
+	// Commands that should skip config loading (don't need API client)
+	skipCommands := []struct {
+		name       string
+		cmdPath    string
+		cmdName    string
+		shouldSkip bool
+	}{
+		// Commands that SHOULD skip config loading
+		{"config command", "homeyctl config", "config", true},
+		{"config set-token", "homeyctl config set-token", "set-token", true},
+		{"config set-host", "homeyctl config set-host", "set-host", true},
+		{"config show", "homeyctl config show", "show", true},
+		{"version command", "homeyctl version", "version", true},
+		{"help command", "homeyctl help", "help", true},
+		{"completion command", "homeyctl completion", "completion", true},
+		{"ai command", "homeyctl ai", "ai", true},
+		{"login command", "homeyctl login", "login", true},
+		{"token create command", "homeyctl token create", "create", true},
+		{"token scopes command", "homeyctl token scopes", "scopes", true},
+		{"root command", "homeyctl", "homeyctl", true},
+
+		// Commands that should NOT skip config loading (need API client)
+		// This is the key fix for GitHub issues #4 and #5
+		{"flows create command", "homeyctl flows create", "create", false},
+		{"flows create --advanced", "homeyctl flows create", "create", false},
+		{"flows list command", "homeyctl flows list", "list", false},
+		{"flows get command", "homeyctl flows get", "get", false},
+		{"flows update command", "homeyctl flows update", "update", false},
+		{"flows delete command", "homeyctl flows delete", "delete", false},
+		{"flows trigger command", "homeyctl flows trigger", "trigger", false},
+		{"flows cards command", "homeyctl flows cards", "cards", false},
+		{"devices list command", "homeyctl devices list", "list", false},
+		{"devices get command", "homeyctl devices get", "get", false},
+		{"devices set command", "homeyctl devices set", "set", false},
+		{"zones list command", "homeyctl zones list", "list", false},
+		{"token list command", "homeyctl token list", "list", false},
+		{"token delete command", "homeyctl token delete", "delete", false},
+	}
+
+	for _, tc := range skipCommands {
+		t.Run(tc.name, func(t *testing.T) {
+			shouldSkip := shouldSkipConfigLoading(tc.cmdPath, tc.cmdName)
+			if shouldSkip != tc.shouldSkip {
+				t.Errorf("shouldSkipConfigLoading(%q, %q) = %v, want %v",
+					tc.cmdPath, tc.cmdName, shouldSkip, tc.shouldSkip)
+			}
+		})
+	}
+}
+
+// TestFlowsCreateRequiresClient specifically tests the fix for GitHub issues #4 and #5
+// The bug was that cmd.Name() == "create" matched both "token create" and "flows create",
+// causing apiClient to be nil for flows create, resulting in a segmentation fault.
+func TestFlowsCreateRequiresClient(t *testing.T) {
+	// These two commands both have name "create" but only token create should skip
+	tokenCreate := shouldSkipConfigLoading("homeyctl token create", "create")
+	flowsCreate := shouldSkipConfigLoading("homeyctl flows create", "create")
+
+	if !tokenCreate {
+		t.Error("token create should skip config loading (handles its own OAuth)")
+	}
+	if flowsCreate {
+		t.Error("flows create should NOT skip config loading (needs apiClient)")
+	}
+}
+
+// shouldSkipConfigLoading mirrors the logic in PersistentPreRunE
+// This allows us to test the logic without executing actual commands
+func shouldSkipConfigLoading(cmdPath, cmdName string) bool {
+	// Skip config commands that don't need API client
+	if cmdName == "config" || cmdName == "version" || cmdName == "help" ||
+		cmdName == "set-token" || cmdName == "set-host" || cmdName == "show" ||
+		cmdName == "completion" || cmdName == "ai" || cmdName == "scopes" ||
+		cmdName == "login" || cmdPath == "homeyctl" {
+		return true
+	}
+
+	// Only skip create for "token create" (handles its own OAuth auth)
+	// NOT for "flows create" which needs the normal apiClient
+	if cmdPath == "homeyctl token create" {
+		return true
+	}
+
+	return false
+}

